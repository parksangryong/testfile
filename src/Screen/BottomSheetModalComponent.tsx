import React, { useMemo, useCallback } from 'react';
import { Platform } from 'react-native';
// bottom-sheet
import { BottomSheetModal, BottomSheetBackdrop } from '@gorhom/bottom-sheet';

const BottomSheetModalComponent = ({
  bottomSheetModalRef,
  children,
  iosSnap,
  andSnap,
}: any) => {
  /** 바텀시트 높이 */
  const snapPoints = useMemo(
    () => (Platform.OS === 'ios' ? [iosSnap] : [andSnap]),
    [],
  );

  /** 백드롭 */
  const renderBackdrop = useCallback(
    (props: any) => (
      <BottomSheetBackdrop
        {...props}
        pressBehavior="close"
        appearsOnIndex={0}
        disappearsOnIndex={-1}
        opacity={0.8}
      />
    ),
    [],
  );

  return (
    <BottomSheetModal
      ref={bottomSheetModalRef}
      index={0}
      snapPoints={snapPoints}
      backdropComponent={renderBackdrop}
    >
      {children}
    </BottomSheetModal>
  );
};

export default BottomSheetModalComponent;
