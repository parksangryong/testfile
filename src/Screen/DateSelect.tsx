import React, { useRef, useState } from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

//style
import { COLORS, FONTS } from '../Sub/Constants';

//component
import RenderOptions from './RenderOption';

const DateSelect = ({
  itemHeight,
  pointColor,
  pointBackgroundColor,
  itemFontSize,
  handlePress,
}: any) => {
  const [selectedYear, setSelectedYear] = useState(new Date().getFullYear());
  const [selectedMonth, setSelectedMonth] = useState(new Date().getDay() - 1);
  const [selectedDay, setSelectedDay] = useState(new Date().getDate());

  const yearScrollViewRef = useRef(null);
  const monthScrollViewRef = useRef(null);
  const dayScrollViewRef = useRef(null);

  const years = Array.from(
    { length: 21 },
    (_, index) => new Date().getFullYear() - 10 + index,
  );
  const months = Array.from({ length: 12 }, (_, index) => index + 1);
  const days = Array.from(
    { length: new Date(selectedYear, selectedMonth, 0).getDate() },
    (_, index) => index + 1,
  );

  return (
    <View style={styles.container}>
      <View style={styles.dateBox}>
        <View
          style={[
            styles.activeBar,
            // eslint-disable-next-line react-native/no-inline-styles
            {
              height: itemHeight ? itemHeight : 40,
              top: itemHeight ? (200 - itemHeight) / 2 : 80,
              backgroundColor: pointBackgroundColor
                ? pointBackgroundColor
                : COLORS.opacityColor.mint,
            },
          ]}
        />
        <RenderOptions
          options={years}
          setSelectedValue={setSelectedYear}
          text="년"
          scrollViewRef={yearScrollViewRef}
          itemHeight={itemHeight}
          itemFontSize={itemFontSize}
          selectedValue={selectedYear}
          pointColor={pointColor}
        />
        <RenderOptions
          options={months}
          setSelectedValue={setSelectedMonth}
          text="월"
          scrollViewRef={monthScrollViewRef}
          itemHeight={itemHeight}
          itemFontSize={itemFontSize}
          selectedValue={selectedMonth}
          pointColor={pointColor}
        />
        <RenderOptions
          options={days}
          setSelectedValue={setSelectedDay}
          text="일"
          scrollViewRef={dayScrollViewRef}
          itemHeight={itemHeight}
          itemFontSize={itemFontSize}
          selectedValue={selectedDay}
          pointColor={pointColor}
        />
      </View>
      <TouchableOpacity
        style={styles.btn}
        onPress={() =>
          handlePress({
            year: selectedYear,
            month: selectedMonth,
            day: selectedDay,
          })
        }
      >
        <Text style={styles.btntext}>선택완료</Text>
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginBottom: 20,
    width: '100%',
    paddingHorizontal: 10,
  },
  dateBox: {
    flex: 2,
    flexDirection: 'row',
  },
  optionContainer: {
    maxHeight: 210,
    width: '30%',
    borderRadius: 5,
    marginVertical: 10,
  },
  option: {
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  btn: {
    width: '100%',
    height: 60,
    backgroundColor: COLORS.defaultColor.main,
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 10,
  },
  btntext: {
    fontFamily: FONTS.bold,
    lineHeight: 60,
    color: COLORS.defaultColor.white,
    fontSize: 16,
  },
  activeBar: {
    width: '100%',
    zIndex: -1,
    position: 'absolute',
    backgroundColor: COLORS.opacityColor.mint,
    top: 80,
    height: 40,
  },
});
export default DateSelect;
