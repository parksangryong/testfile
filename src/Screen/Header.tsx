import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  TouchableOpacity,
  Text,
  StyleSheet,
} from 'react-native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useNavigation } from '@react-navigation/native';

const Header = () => {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();

  return (
    <SafeAreaView>
      <StatusBar barStyle={'dark-content'} />
      <TouchableOpacity
        style={styles.hback}
        onPress={() => navigation.goBack()}
        activeOpacity={1}
      >
        <Text style={styles.hx}>x</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  hback: {
    width: '100%',
    height: 50,
    backgroundColor: 'black',
    justifyContent: 'center',
    alignItems: 'flex-end',
  },

  hx: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 30,
    marginTop: 10,
    marginRight: 10,
  },
});

export default Header;
