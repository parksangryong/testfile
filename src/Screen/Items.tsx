import React, { useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  TextInput,
} from 'react-native';
import {
  usePostMutation,
  usePutMutation,
  useDeleteMutation,
} from '../Sub/query';

const Items = ({ id, name, age }: any) => {
  const postMutation = usePostMutation();
  const putMutation = usePutMutation();
  const deleteMutation = useDeleteMutation();

  const [nameValue, setNameValue] = useState(name);
  const [ageValue, setAgeValue] = useState(age);

  return (
    <>
      <View style={stlyes.Container}>
        <Text style={stlyes.id}>{id}</Text>
        <Text style={stlyes.age}>{age}</Text>
        <Text style={stlyes.name}>{name}</Text>
        <TouchableOpacity
          style={stlyes.btn}
          onPress={() =>
            postMutation.mutate({
              endPoint: 'ddd',
              name: nameValue,
              age: ageValue,
            })
          }
        >
          <Text style={stlyes.btntext}>post</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={stlyes.btn}
          onPress={() =>
            putMutation.mutate({ id: id, name: nameValue, age: ageValue })
          }
        >
          <Text style={stlyes.btntext}>put</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={stlyes.btn}
          onPress={() => deleteMutation.mutate({ id: id })}
        >
          <Text style={stlyes.btntext}>delete</Text>
        </TouchableOpacity>
      </View>
      <View style={stlyes.Container}>
        <Text style={stlyes.label}>이름: </Text>
        <TextInput
          style={stlyes.input}
          value={nameValue}
          onChangeText={data => setNameValue(data)}
        />
        <Text style={stlyes.label}>나이: </Text>
        <TextInput
          style={stlyes.input}
          value={ageValue}
          onChangeText={data => setAgeValue(data)}
        />
      </View>
    </>
  );
};

export default Items;

const stlyes = StyleSheet.create({
  Container: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: 40,
    borderBottomWidth: 1,
  },
  id: {
    color: 'red',
    width: 20,
  },
  name: {
    color: 'blue',
    width: 100,
  },
  age: {
    color: 'green',
    width: 35,
  },
  btn: {
    height: 40,
    width: 46,
    justifyContent: 'center',
    alignItems: 'center',
  },
  btntext: {
    color: 'black',
    textDecorationLine: 'underline',
  },
  input: {
    backgroundColor: 'skyblue',
    width: 150,
    height: 30,
    fontSize: 15,
    lineHeight: 20,
    paddingVertical: 0,
    paddingHorizontal: 10,
    margin: 0,
    color: 'black',
  },
  label: {
    color: 'black',
    fontWeight: '500',
  },
});
