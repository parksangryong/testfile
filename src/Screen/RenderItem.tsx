import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { COLORS, FONTS } from '../Sub/Constants';

const RenderItem = ({
  option,
  itemFontSize,
  itemHeight,
  // setSelectedValue,
  text,
  selectedValue,
  pointColor,
}: any) => {
  return (
    <TouchableOpacity
      activeOpacity={0.7}
      style={[
        styles.option,
        // eslint-disable-next-line react-native/no-inline-styles
        {
          height: itemHeight ? itemHeight : 40,
        },
      ]}
      // onPress={() => setSelectedValue(option)}
    >
      <Text
        style={[
          styles.text,
          // eslint-disable-next-line react-native/no-inline-styles
          {
            fontFamily: option === selectedValue ? FONTS.bold : FONTS.regular,
            color:
              option !== selectedValue
                ? COLORS.defaultColor.black
                : pointColor
                ? pointColor
                : COLORS.defaultColor.main,
            fontSize: itemFontSize ? itemFontSize : 17,
            lineHeight: itemFontSize ? itemFontSize + 5 : 22,
          },
        ]}
      >
        {option} {text}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  option: {
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 17,
    lineHeight: 22,
    color: COLORS.defaultColor.deepGray,
    fontFamily: FONTS.regular,
    padding: 0,
  },
});

export default RenderItem;
