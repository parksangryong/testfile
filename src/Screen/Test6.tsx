import React from 'react';
import { SafeAreaView, View, Text, StyleSheet, ScrollView } from 'react-native';

export const Test6 = () => {
  return (
    <SafeAreaView style={styles.body}>
      <ScrollView style={styles.container}>
        <View style={styles.list}>
          <Text>Title</Text>
        </View>
        <View style={[styles.list, styles.textbox]}>
          <Text>Text</Text>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    margin: 15,
    flex: 1,
    borderRadius: 10,
    padding: 15,
    borderColor: 'gray',
  },
  body: {
    width: '100%',
    height: '100%',
    backgroundColor: 'white',
  },
  list: {
    borderWidth: 1,
    height: 40,
    borderRadius: 15,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 15,
    borderColor: 'gray',
  },
  textbox: {
    height: 600,
    marginTop: 20,
  },
});
