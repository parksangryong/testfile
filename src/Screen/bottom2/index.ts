export { default as VTest1 } from './VTest1';
export { default as VTest2 } from './VTest2';
export { default as VTest3 } from './VTest3';
