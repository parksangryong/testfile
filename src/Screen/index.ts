export { default as Test1 } from './Test1';
export { default as Test2 } from './Test2';
export { default as Test3 } from './Test3';
export { default as Test4 } from './Test4';
export { default as Test5 } from './Test5';
export { default as Test6 } from './Test6';

export { default as Top1 } from './Top1';
export { default as Top2 } from './Top2';
export { default as Top3 } from './Top3';
export { default as Top4 } from './Top4';
export { default as Top5 } from './Top5';
export { default as Top6 } from './Top6';
export { default as Top7 } from './Top7';

export { default as Videos } from './Videos';
export { default as Header } from './Header';
export { default as TopHeader } from './TopHeader';

export { default as BottomSheetModalComponent } from './BottomSheetModalComponent';
export { default as DateSelect } from './DateSelect';
