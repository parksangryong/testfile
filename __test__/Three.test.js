import React from 'react';
import 'react-native';
import { render, screen } from '@testing-library/react-native';
import { Test6 } from '../src/Screen/Test6'; //컴포넌트 예시

describe('Test3', () => {
  it('Home 화면이 정상적으로 렌더링 되는가?', () => {
    // 목적과 콜백함수 (테스트)
    const screens = render(<Test6 />);
    const json = screens.toJSON(); // 렌더링 된 컴포넌트를 json 구조로 변환한다.
    expect(json).toMatchSnapshot(); // 만들어진 json으로 스냅샷을 촬영한다.
  });
  it('title이 존재하는가?', () => {
    render(<Test6 />);
    const leng = screen.getAllByText('Title');
    expect(leng.length).toEqual(1);
  });
  it('text가 존재하는가?', () => {
    render(<Test6 />);
    const leng = screen.getAllByText('Text');
    expect(leng.length).toEqual(1);
  });
}); // 대 주제 생성
