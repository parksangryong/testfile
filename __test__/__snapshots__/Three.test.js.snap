// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test3 Home 화면이 정상적으로 렌더링 되는가? 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    style={
      {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "margin": 15,
        "padding": 15,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "gray",
            "borderRadius": 15,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "paddingHorizontal": 15,
          }
        }
      >
        <Text>
          Title
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "gray",
              "borderRadius": 15,
              "borderWidth": 1,
              "height": 40,
              "justifyContent": "center",
              "paddingHorizontal": 15,
            },
            {
              "height": 600,
              "marginTop": 20,
            },
          ]
        }
      >
        <Text>
          Text
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
